### PR TITLE
Add broadcast operation as utility

### PR DESCRIPTION
**Pull request by huningxin**

### CODE REVIEW

**Comment by huningxin** [comment]
@BruceDai, PTAL. The broadcast op could help binary implementation.

File: src/broadcast.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  const newRank = newShape.length;
+  if (newRank < input.rank) {
+    throw new Error(`The rank of new shape ${newRank} is invalid.`);
+  }
+  const broadcastAxes = new Array(input.rank).fill(false);
+  for (let i = 0; i < input.rank; ++i) {
+    const newAxis = newRank - i - 1;
+    const axis = input.rank - i - 1;
+    if (input.shape[axis] === 1 && newShape[newAxis] !== 1) {
+      broadcastAxes[newAxis] = true;
```
Comment (BruceDai) [inline src/broadcast.js:19]: Here `broadcastAxes[newAxis]` should be `broadcastAxes[axis]`.

**Review by BruceDai — COMMENTED** [review]
@huningxin There is a typos, please fix it, thanks.

File: src/broadcast.js (hunk 1 above)
Comment (huningxin) [reply src/broadcast.js]: good catch, thanks @BruceDai . I fixed it in the new commit, please take a look.

Comment (BruceDai) [reply src/broadcast.js]: LGTM, thanks.

**Review by BruceDai — APPROVED** [review]

**Pull request merged**